### ISSUE TITLE
Proposal: Keyboard driven piano roll mode

### ISSUE THREAD

**Issue by jneen** [LMMS/lmms#6387]
### Enhancement Summary

A keyboard mode for editing in the piano roll - either enabled by default, turned on via config, or a specific tool.

* Includes a "point" in the pitch/time grid, movable by simple arrow keys
* Inputting a note via MIDI or the computer keyboard would result in a new note at the current "point"
* Ability to "grab" notes and move/resize/adjust parameters with modifier keys
* Keyboard-driven selection / multi-note editing

#### Justification

* Accessibility for people with limited ability to use the mouse
* Faster and more accurate note-programming workflow
* Faster and more accurate editing interface
* The "point" could be used as a default location for other features, and is fairly intuitive as a "place where *you* are" marker

#### Mockup

If there is interest, I will work on a mockup specifically with the LMMS piano roll, including choosing specific keys for specific functions, but in the meantime I have a general interactive demo here:

http://files.jneen.net/ui.html

**Comment by jneen** [comment]
(I should note, I'm willing to code, test, and iterate on this)

**Comment by Spekular** [comment]
The existing step recording feature seems related: 
https://github.com/LMMS/lmms/pull/4544

**Comment by jneen** [comment]
Yes! It would ideally be an enhancement to that feature, adding things like modifying, moving, and deleting notes after they have already been input.
